### PR TITLE
Plasma 5.25

### PR DESCRIPTION
**Pull request by hamburger1984**
###### Description of changes

KDE released Plasma [5.25](https://kde.org/announcements/plasma/5/5.25.0/) today 

_Old description:_

> KDE released Plasma [5.25 Beta](https://kde.org/announcements/plasma/5/5.24.90/)
> 
> Let's use this to test it and get fixes in place to be ready for the release.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/desktops/plasma-5/kwin/default.nix [hunk 1]
```diff
@@ -2,17 +2,19 @@
   mkDerivation, lib,
   extra-cmake-modules, kdoctools, fetchpatch,
 
-  libepoxy, lcms2, libICE, libSM, libcap, libdrm, libinput, libxkbcommon, mesa,
-  pipewire, udev, wayland, xcb-util-cursor, xwayland,
+  libepoxy, lcms2, libICE, libSM, libcap, libdrm, libinput, libxkbcommon,
+  libxcvt, mesa, pipewire, plasma-wayland-protocols, udev, wayland,
+  wayland-protocols, xcb-util-cursor, xwayland,
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/kwin/default.nix:7]: Please don't reflow this. It makes the diff unredable.

File: pkgs/desktops/plasma-5/kwin/default.nix [hunk 2]
```diff
@@ -21,20 +23,27 @@ mkDerivation {
   pname = "kwin";
   nativeBuildInputs = [ extra-cmake-modules kdoctools ];
   buildInputs = [
-    libepoxy lcms2 libICE libSM libcap libdrm libinput libxkbcommon mesa pipewire
-    udev wayland xcb-util-cursor xwayland
+    libepoxy lcms2 libICE libSM libcap libdrm libinput libxkbcommon libxcvt
+    mesa pipewire plasma-wayland-protocols udev wayland wayland-protocols
+    xcb-util-cursor xwayland
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/kwin/default.nix:28]: same here

File: pkgs/desktops/plasma-5/kwin/default.nix (hunk 1 above)
Comment (hamburger1984) [reply pkgs/desktops/plasma-5/kwin/default.nix]: moving all additions to a separate line + keep current formatting

File: pkgs/desktops/plasma-5/kwin/default.nix (hunk 2 above)
Comment (hamburger1984) [reply pkgs/desktops/plasma-5/kwin/default.nix]: moving all additions to a separate line + keep current formatting

**Comment by InternetUnexplorer** [comment]
Thanks for doing this!

It looks like `xdg-desktop-portal-kde` ≥5.24.90 requires `plasma-wayland-protocols` ≥1.7.0; I made a PR to your fork for that.

Other than that, `libkscreen` doesn't seem to build:
```sh
$ nix build github:hamburger1984/nixpkgs/plasma-5.25-beta#plasma5Packages.libkscreen
```
```
/build/libkscreen-5.24.90/src/doctor/dpmsclient.cpp:19:10: fatal error: qwayland-dpms.h: No such file or directory
   19 | #include "qwayland-dpms.h"
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```
I can't figure out which dependency is supposed to provide that (maybe `qtwayland`? but I'm not familiar with how the Qt stuff is packaged), so I thought I'd leave it here in case anyone else has ideas.

File: pkgs/desktops/plasma-5/kwin/default.nix [hunk 3]
```diff
@@ -23,18 +25,25 @@ mkDerivation {
...
-    kservice kwayland kwayland-server kwidgetsaddons kwindowsystem kxmlgui
+    kservice kwayland kwidgetsaddons kwindowsystem kxmlgui
     plasma-framework libqaccessibilityclient
 
   ];
   outputs = [ "out" "dev" ];
+
+  postPatch = ''
+    find . -type f -name "*.py" \
+      -exec sed -i 's,#!/usr/bin/env python3,#!${python3}/bin/python,' {} \;
```
Comment (SuperSandro2000) [inline pkgs/desktops/plasma-5/kwin/default.nix:44]: We can't use patchShebangs, can we?

Comment (hamburger1984) [reply pkgs/desktops/plasma-5/kwin/default.nix:44]: Sure we can - I was suspecting something like this to exist, but could not find it. Will have a look at other usages of "patchShebangs" and adjust this.

**Comment by hamburger1984** [comment]
> Thanks for doing this!
> 
> It looks like `xdg-desktop-portal-kde` ≥5.24.90 requires `plasma-wayland-protocols` ≥1.7.0; I made a PR to your fork for that.
> 
> Other than that, `libkscreen` doesn't seem to build:
> 
> ```shell
> $ nix build github:hamburger1984/nixpkgs/plasma-5.25-beta#plasma5Packages.libkscreen
> ```
> 
> ```
> /build/libkscreen-5.24.90/src/doctor/dpmsclient.cpp:19:10: fatal error: qwayland-dpms.h: No such file or directory
>    19 | #include "qwayland-dpms.h"
>       |          ^~~~~~~~~~~~~~~~~
> compilation terminated.
> ```
> 
> I can't figure out which dependency is supposed to provide that (maybe `qtwayland`? but I'm not familiar with how the Qt stuff is packaged), so I thought I'd leave it here in case anyone else has ideas.

I merged your PR into this one. Will have another look at this whole topic later today or during the weekend.

File: pkgs/desktops/plasma-5/kwin/default.nix [hunk 4]
```diff
@@ -23,18 +25,24 @@ mkDerivation {
...
+    kservice kwayland kwidgetsaddons kwindowsystem kxmlgui
     plasma-framework libqaccessibilityclient
 
   ];
   outputs = [ "out" "dev" ];
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
```
Comment (hamburger1984) [inline pkgs/desktops/plasma-5/kwin/default.nix:45]: Much cleaner - thanks for the hint, @SuperSandro2000

File: pkgs/desktops/plasma-5/kwin/default.nix [hunk 5]
```diff
@@ -23,18 +25,24 @@ mkDerivation {
...
-    kservice kwayland kwayland-server kwidgetsaddons kwindowsystem kxmlgui
+    kservice kwayland kwidgetsaddons kwindowsystem kxmlgui
     plasma-framework libqaccessibilityclient
 
   ];
   outputs = [ "out" "dev" ];
+
+  postPatch = ''
+    patchShebangs src/effects/strip-effect-metadata.py
+  '';
```
Comment (hamburger1984) [inline pkgs/desktops/plasma-5/kwin/default.nix:44]: Now we're just patching the one python file that was causing issues during build. 
CC @SuperSandro2000?

File: pkgs/desktops/plasma-5/kwin/default.nix (hunk 4 above)
Comment (hamburger1984) [reply pkgs/desktops/plasma-5/kwin/default.nix]: You're right. It's just one python file. Fixing and (force) pushing ;-)

File: pkgs/desktops/plasma-5/kwin/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/desktops/plasma-5/kwin/default.nix:44]: 👍🏼

**Comment by K900** [comment]
The final release is now up: https://kde.org/announcements/plasma/5/5.25.0/

**Comment by hamburger1984** [comment]
Will update tonight. (If no one else is faster 😉)

**Comment by andrevmatos** [comment]
It seems there's a `qdbus` path being pulled from the wrong derivation:
```
Jun 14 15:29:38 midichlorian systemd[2103]: Starting KDE Configuration Module Initialization (Phase 1)...
Jun 14 15:29:38 midichlorian systemd[2641]: plasma-kcminit-phase1.service: Failed to locate executable /nix/store/qfbzglp8qzvpwyp2pq75s1jsqgjm15sz-qtbase-5.15.3-dev/bin/qdbus: No such file or directory
Jun 14 15:29:38 midichlorian systemd[2641]: plasma-kcminit-phase1.service: Failed at step EXEC spawning /nix/store/qfbzglp8qzvpwyp2pq75s1jsqgjm15sz-qtbase-5.15.3-dev/bin/qdbus: No such file or directory
Jun 14 15:29:38 midichlorian systemd[2103]: plasma-kcminit-phase1.service: Main process exited, code=exited, status=203/EXEC
Jun 14 15:29:38 midichlorian systemd[2103]: plasma-kcminit-phase1.service: Failed with result 'exit-code'.
Jun 14 15:29:38 midichlorian systemd[2103]: Failed to start KDE Configuration Module Initialization (Phase 1).
```

**Comment by andrevmatos** [comment]
It seems the above is caused by `plasma-workspace` having changed its `query_qmake` call with `ecm_query_qt` in `CMakeLists.txt`. Adjusting `postPatch` should be enough.

**Comment by hamburger1984** [comment]
> It seems the above is caused by `plasma-workspace` having changed its `query_qmake` call with `ecm_query_qt` in `CMakeLists.txt`. Adjusting `postPatch` should be enough.

that change already happened in 5.24. good catch. build is currently running locally.

**Comment by hamburger1984** [comment]
I'm getting a segfault trying to open `systemsettings kcm_networkmanagement`. Can anybody confirm?

**Comment by SuperSandro2000** [comment]
> should we retarget this to staging? or is master fine for this update?

Master is fine like always with KDE. Same for frameworks because rebuild amounts get bloated by older qt5 releases.

**Comment by peterhoeg** [comment]
> We should add a wrapper or set an env to fix this for everyone.

It's not that easy unfortunately as a lot of different KDE components exhibit this problem with a stale ~/.cache/something/qmlcache directory. I'll open a separate issue for this.

**Comment by hamburger1984** [comment]
> > I'm getting a segfault trying to open `systemsettings kcm_networkmanagement`. Can anybody confirm?
> > Nevermind - this was easily fixed with a `rm -rf ~/.cache/systemsettings` (got that hint from #125545)
> 
> We should add a wrapper or set an env to fix this for everyone.

Just for clarification: That cache clearing had to be done just once. Not on every start of the systemsettings module.

Can you assist with that? tbh I don't have a clue about that part (yet).

**Comment by peterhoeg** [comment]
And here it is: https://github.com/NixOS/nixpkgs/issues/177720

**Comment by peterhoeg** [comment]
Just to make it clear - the qmlcache issue is *not* a blocker for merging this so let's get it out there!

**Comment by SuperSandro2000** [comment]
Yeah, the caching problem is already present since months if not longer.

**Comment by peterhoeg** [comment]
> Yeah, the caching problem is already present since months if not longer.

Since 2017!

**Pull request merged**